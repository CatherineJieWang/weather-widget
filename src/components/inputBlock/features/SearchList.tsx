import React, { FC } from "react";
import { SearchResult, Location } from "../types";
import "../styled/searchlist.scss";
import { getWeather } from "../../../data/fetchDataFromServer";

const SearchList: FC<SearchResult> = (props) => {
  const { searchResult,setInputValue, setCity, setOpen, setWeathers } = props;
  const getWeatherFromServer = async (cityId: number) => {
    try {
      if (cityId) {
        const weatherResult = await getWeather(cityId);
        if (weatherResult.data) {
          setWeathers(weatherResult.data.slice(0, 5));
        }
      }
    } catch {
      console.log("Server Error");
    }
  };

  return (
    <div className="search-container">
      {searchResult &&
        searchResult.map((data: Location, index: number) => (
          <p
            className="search-container__row"
            key={index}
            onClick={() => {
              if (data.woeid) getWeatherFromServer(data.woeid);
              setInputValue('')
              setCity(data.title);
              setOpen(false);
            }}
          >
            {data.title}
          </p>
        ))}
    </div>
  );
};

export default SearchList;
