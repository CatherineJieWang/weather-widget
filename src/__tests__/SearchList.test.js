import React from "react";
import renderer from "react-test-renderer";
import SearchList from "../components/inputBlock/features/SearchList";

test("<SearchList /> renders with correct result with 'Sant' as prop", () => {
  const searchResult = [
    {
      title: "Santa Cruz",
      location_type: "City",
      woeid: 2488853,
      latt_long: "36.974018,-122.030952",
    },
    {
      title: "Santiago",
      location_type: "City",
      woeid: 349859,
      latt_long: "-33.463039,-70.647942",
    },
    {
      title: "Santorini",
      location_type: "City",
      woeid: 56558361,
      latt_long: "36.406651,25.456530",
    },
    {
      title: "Santander",
      location_type: "City",
      woeid: 773964,
      latt_long: "43.461498,-3.810010",
    },
    {
      title: "Santa Cruz de Tenerife",
      location_type: "City",
      woeid: 773692,
      latt_long: "28.46163,-16.267059",
    },
    {
      title: "Santa Fe",
      location_type: "City",
      woeid: 2488867,
      latt_long: "35.666431,-105.972572",
    },
  ];
  const tree = renderer
    .create(
      <SearchList
        searchResult={searchResult}
        setInputValue={() => console.log("setInputValue")}
        setCity={() => console.log("setCity")}
        setOpen={() => console.log("setOpen")}
        setWeathers={() => console.log("setWeathers")}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
