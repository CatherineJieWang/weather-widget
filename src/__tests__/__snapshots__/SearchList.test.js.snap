// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchList /> renders with correct result with 'Sant' as prop 1`] = `
<div
  className="search-container"
>
  <p
    className="search-container__row"
    onClick={[Function]}
  >
    Santa Cruz
  </p>
  <p
    className="search-container__row"
    onClick={[Function]}
  >
    Santiago
  </p>
  <p
    className="search-container__row"
    onClick={[Function]}
  >
    Santorini
  </p>
  <p
    className="search-container__row"
    onClick={[Function]}
  >
    Santander
  </p>
  <p
    className="search-container__row"
    onClick={[Function]}
  >
    Santa Cruz de Tenerife
  </p>
  <p
    className="search-container__row"
    onClick={[Function]}
  >
    Santa Fe
  </p>
</div>
`;
